src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -26,11 +26,21 @@ public abstract class AssemblyLoadContext
         [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
         [SuppressUnmanagedCodeSecurity]
         private static extern bool CanUseAppPathAssemblyLoadContextInCurrentDomain();
-        
+
+#if !FEATURE_COLLECTIBLE_ALC
         [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
         [SuppressUnmanagedCodeSecurity]
         private static extern IntPtr InitializeAssemblyLoadContext(IntPtr ptrAssemblyLoadContext);
-        
+#else // !FEATURE_COLLECTIBLE_ALC
```

Comment:
You can alleviate the need for two definitions for InitializeAssemblyLoadContext by keeping only the one with fIsCollectible and update all the callsites to pass "false" for fIsCollectivel when FEATURE_COLLECTIVE_ALC is not defined.

This helps make the code more readable.